kevm-pyk/src/kevm_pyk/utils.py

```diff
@@ -168,6 +168,61 @@ def kevm_prove(
         return False
 
 
+def kevm_debug(
+    kprove: KProve,
+    proof: Proof,
+    kcfg_explore: KCFGExplore,
+    max_depth: int = 1000,
+    max_iterations: int | None = None,
+    break_every_step: bool = False,
+    break_on_jumpi: bool = False,
+    break_on_calls: bool = True,
+    extract_branches: Callable[[CTerm], Iterable[KInner]] | None = None,
+    abstract_node: Callable[[CTerm], CTerm] | None = None,
+) -> Prover:
+    terminal_rules = ['EVM.halt']
+    cut_point_rules = []
+    if break_every_step:
+        terminal_rules.append('EVM.step')
+    if break_on_jumpi:
+        cut_point_rules.extend(['EVM.jumpi.true', 'EVM.jumpi.false'])
+    if break_on_calls:
+        cut_point_rules.extend(
+            [
+                'EVM.call',
+                'EVM.callcode',
+                'EVM.delegatecall',
+                'EVM.staticcall',
+                'EVM.create',
+                'EVM.create2',
+                'FOUNDRY.foundry.call',
+                'EVM.end',
+                'EVM.return.exception',
+                'EVM.return.revert',
+                'EVM.return.success',
+            ]
+        )
```

Comment:
I am not entirely sure that duplicating this code is a good idea. Perhaps you could extract this into its own function and use it both here and in `kevm_prove`. 

Something along the lines of :
```python
def generate_rules(break_every_step: bool, break_on_jumpi: bool, break_on_calls: bool):
    terminal_rules = ['EVM.halt']
    cut_point_rules = []
    if break_every_step:
        terminal_rules.append('EVM.step')
    if break_on_jumpi:
        cut_point_rules.extend(['EVM.jumpi.true', 'EVM.jumpi.false'])
    if break_on_calls:
        cut_point_rules.extend(
            [
                'EVM.call',
                'EVM.callcode',
                'EVM.delegatecall',
                'EVM.staticcall',
                'EVM.create',
                'EVM.create2',
                'FOUNDRY.foundry.call',
                'EVM.end',
                'EVM.return.exception',
                'EVM.return.revert',
                'EVM.return.success',
            ]
        )
    return terminal_rules, cut_point_rules
```

Reply:
Is this code being duplicated? It looks like it's being moved to here. Maybe this comment is resolved?

Reply:
Yes, I followed Andrei's recommendation and refactored this part.